src/components/SarStakingWidget/Stake/DrawerContent/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { getEtherscanLink } from 'src/utils';
+import { Box, Button, CurrencyLogo, Loader, Stat, Text } from '../../..';
+import { ErrorBox, ErrorWrapper, Footer, Header, Link, Root, SubmittedWrapper, TokenRow } from './styled';
+
+interface Props {
+  stakeAmount?: CurrencyAmount;
+  token: Token;
+  dollerWorth?: number;
+  weeklyPNG: BigNumber;
+  attemptingTxn: boolean;
+  txHash: string | null;
+  onConfirm: () => void;
+  errorMessage: string | null;
+  onClose: () => void;
+}
+
+const DrawerContent: React.FC<Props> = (props) => {
+  const { stakeAmount, token, dollerWorth, weeklyPNG, attemptingTxn, errorMessage, txHash, onClose, onConfirm } = props;
+
+  const { chainId } = usePangolinWeb3();
+  const theme = useContext(ThemeContext);
+
+  const { t } = useTranslation();
+
+  const ConfirmContent = (
+    <Root paddingX="20px" paddingBottom="20px">
+      <Header>
+        <TokenRow>
+          <Text fontSize={24} fontWeight={500} color="text1" style={{ marginRight: '12px' }}>
+            {stakeAmount?.toSignificant(6) ?? 0}
+          </Text>
+          <CurrencyLogo currency={token} size={24} imageSize={48} />
+        </TokenRow>
+        <Box display="inline-grid" style={{ gridGap: '10px', gridTemplateColumns: 'auto auto' }}>
+          <Stat
+            title={t('sarStake.dollarValue')}
+            titlePosition="top"
+            stat={`${dollerWorth ?? 0}$`}
+            titleColor="text2"
+          />
+          <Stat title={t('sarStake.startingApr')} titlePosition="top" stat={'0%'} titleColor="text2" />
+        </Box>
+        <Box display="flex" flexDirection="row" justifyContent="space-between">
+          <Text color="text1">{t('sarStake.weeklyDistributed', { symbol: token.symbol })}</Text>
+          <Text color="text1">{numeral(formatEther(weeklyPNG)).format('0.00a')}</Text>
+        </Box>
+        <Box bgColor="color3" borderRadius="4px" padding="15px">
+          <Text color="text1" fontWeight={400} fontSize="14px" textAlign="center">
+            {t('sarStake.confirmDescription')}
+          </Text>
+        </Box>
+      </Header>
+      <Footer>
+        <Box my={'10px'}>
+          <Button variant="primary" onClick={onConfirm}>
+            {t('sarStake.stake')}
+          </Button>
+        </Box>
+      </Footer>
+    </Root>
+  );
+
+  const PendingContent = (
+    <Loader
+      size={100}
+      label={t('sarStake.pending', { balance: stakeAmount?.toSignificant(2) ?? 0, symbol: token.symbol })}
+    />
+  );
+
+  const ErroContent = (
+    <ErrorWrapper>
+      <ErrorBox>
+        <AlertTriangle color={theme.red1} style={{ strokeWidth: 1.5 }} size={64} />
+        <Text fontWeight={500} fontSize={16} color={'red1'} style={{ textAlign: 'center', width: '85%' }}>
+          {errorMessage}
+        </Text>
+      </ErrorBox>
+      <Button variant="primary" onClick={onClose}>
+        {t('transactionConfirmation.dismiss')}
+      </Button>
+    </ErrorWrapper>
+  );
+
+  const SubmittedContent = (
```

Comment:
In all `DrawerContent` component, `ErrorContent` & `SubmittedContent` are duplicated with little to no changes. Please check how we have created this component `src/components/TransactionCompleted/index.tsx` . Try to make same component or reuse it for `SubmittedContent`. Also do same for `ErrorContent`.

Once you make reusable components for this, please make sure to clean up styles related to this.